app/uk/gov/hmrc/filetransmission/services/TransmissionRequestProcessingJob.scala

```diff
@@ -41,23 +44,25 @@ class TransmissionRequestProcessingJob @Inject()(
     implicit val hc = HeaderCarrier()
 
     for (result <- mdgConnector.requestTransmission(item.request)) yield {
-      logResult(item, result)
-      if (Instant.now().isAfter(timeToGiveUp)) {
-        Logger.warn(s"Failed to deliver notification within delivery window for file reference: [${item.request.file.reference}] before [$timeToGiveUp].")
-      }
-      result match {
-        case MdgRequestSuccessful =>
-          callbackSender.sendSuccessfulCallback(item.request)
-          ProcessingSuccessful
-        case MdgRequestFatalError(error) =>
-          callbackSender.sendFailedCallback(item.request, error)
-          ProcessingFailedDoNotRetry(error)
-        case MdgRequestError(error) => {
-          if (nextRetryTime < timeToGiveUp) {
-            ProcessingFailed(error)
-          } else {
+      withLoggedContext(item.request) {
+        logResult(item, result)
+        if (clock.instant().isAfter(timeToGiveUp)) {
```

Comment:
I realise that this logic is not changed in this commit/PR, however I was wondering if the logic is correct given the timeout only results in a warning being logged?

Reply:
Good question. I think we're ok because:
a) in the event of a timeout we still notify the calling service that the message was undelivered. So the knowledge of what happen isn't just hidden in the log message.
b) our assumption is that it's going to be "pseudo normal" practice for some messages to be undelivered. And as long as we've handled that use case as expected then I think it's correct to log a warning rather than an error - since our service is behaving correctly in the given scenario.

Reply:
👍